pkg/runner/cluster_k8s.go

```diff
@@ -293,138 +293,60 @@ func (*ClusterK8sRunner) ID() string {
 	return "cluster:k8s"
 }
 
-func (c *ClusterK8sRunner) healthcheckK8s() (k8sCheck api.HealthcheckItem) {
-	k8sCheck = api.HealthcheckItem{Name: "k8s", Status: api.HealthcheckStatusOK, Message: "k8s cluster is running"}
-	cmd := exec.Command("kops", "validate", "cluster")
-	if err := cmd.Run(); err != nil {
-		out, _ := cmd.CombinedOutput()
-		k8sCheck = api.HealthcheckItem{Name: "k8s", Status: api.HealthcheckStatusFailed, Message: fmt.Sprintf("k8s cluster validation failed: %s; output from kops: %s", err, string(out))}
-		return
-	}
-	return
-}
-
-func (c *ClusterK8sRunner) healthcheckEFS() (efsCheck api.HealthcheckItem) {
-	efsCheck = api.HealthcheckItem{Name: "efs", Status: api.HealthcheckStatusFailed, Message: "efs provisioner is not running"}
-
-	client := c.pool.Acquire()
-	defer c.pool.Release(client)
-
-	pods, err := client.CoreV1().Pods(c.config.Namespace).List(metav1.ListOptions{
-		LabelSelector: "app=efs-provisioner",
-	})
-	if err != nil {
-		efsCheck.Message = err.Error()
-		return
-	}
-	if len(pods.Items) != 1 {
-		efsCheck.Message = fmt.Sprintf("expected 1 EFS provisioner pod. found %d.", len(pods.Items))
-		return
-	}
-
-	pod := pods.Items[0]
-	if pod.Status.Phase != "Running" {
-		return
-	}
-
-	efsCheck = api.HealthcheckItem{Name: "efs", Status: api.HealthcheckStatusOK, Message: "efs provisioner is running"}
-	return
-}
-
-func (c *ClusterK8sRunner) healthcheckRedis() (redisCheck api.HealthcheckItem) {
-	redisCheck = api.HealthcheckItem{Name: "redis", Status: api.HealthcheckStatusFailed, Message: "redis service is not running"}
-
-	client := c.pool.Acquire()
-	defer c.pool.Release(client)
-
-	pods, err := client.CoreV1().Pods(c.config.Namespace).List(metav1.ListOptions{
-		LabelSelector: "app=redis",
-	})
-	if err != nil {
-		redisCheck.Message = err.Error()
-		return
-	}
-	if len(pods.Items) != 1 {
-		redisCheck.Message = fmt.Sprintf("expected 1 redis pod. found %d.", len(pods.Items))
-		return
-	}
-
-	for _, pod := range pods.Items {
-		if pod.Status.Phase != "Running" {
-			return
-		}
-	}
-
-	redisCheck = api.HealthcheckItem{Name: "redis", Status: api.HealthcheckStatusOK, Message: "redis service is running"}
-	return
-}
-
-func (c *ClusterK8sRunner) healthcheckSidecar() (sidecarCheck api.HealthcheckItem) {
-	sidecarCheck = api.HealthcheckItem{Name: "sidecar", Status: api.HealthcheckStatusFailed, Message: "sidecar service is not reachable"}
+func (c *ClusterK8sRunner) Healthcheck(ctx context.Context, engine api.Engine, ow *rpc.OutputWriter, fix bool) (*api.HealthcheckReport, error) {
+	// TODO how does one pass the context to k8s API calls?
```

Comment:
I don't think you can send `ctx` to k8s API calls, I think you can only set a `timeout` param. Would be interested to hear if you find anything.